lib/registry-sync/mirror-source.ts

```diff
@@ -65,26 +65,41 @@ export abstract class MirrorSource {
    *
    * @param image e.g jsii/superchain
    * @param tag optional, defaults to 'latest'
+   *
+   * @deprecated This method's name inaccurately expresses that the image comes
+   * from DockerHub, when any publicly-accessible repository can be used. Prefer
+   * using `fromImageName(string, string?)` instead, which is more aptly named.
    */
   public static fromDockerHub(image: string, tag: string = 'latest'): MirrorSource {
+    return this.fromImageName(image, tag);
+  }
+
+  /**
+   * Configure an image from DockerHub or a repository-qualified image name.
+   *
+   * @param image e.g public.ecr.aws/jsii/superchain
+   * @param tag optional, defaults to 'latest'
+   * @param ecrRepositoryName the name of the ECR Repository to use (e.g: jsii/superchain)
+   */
+  public static fromImageName(image: string, tag: string = 'latest', ecrRepositoryName: string = image.includes('/') ? image : `library/${image}`): MirrorSource {
```

Comment:
```suggestion
  public static fromImageName(image: string, tag: string = 'latest', ecrRepositoryName?: string): MirrorSource {
```

Is this really needed? I see the constructor already defaults `ecrRepositoryName` to `repositoryName`, which seems like that results in the same thing?

Reply:
Yep - the default `library/` prefix is only actioned on that code path, if no custom name is provided... The other cases just use the exact same image name that is provided by default... This was warranted to not change the behavior of existing unit tests.